### ISSUE TITLE
Gas estimation problem buying all tokens

### ISSUE THREAD

**Issue by anxolin** [gnosis/gnosis.js#99]
Hi!

I've followed the SDK tutorial and I think there might be an issue with gas estimation for the `butAllTokens` operation.

The tutorial I was following was:
https://gnosis.github.io/gnosis.js/tutorial-events-oracles-and-markets.html

So I've made a demo web app out of the tutorial:
* Github code: https://github.com/anxolin/gnosis-sdk-example
* Demo: https://wizardly-pasteur-5717ab.netlify.com/

To reproduce the issue, you can use the demo with metamask connected to **Kovan** network.

Then follow the steps:
1. Publish event
2. Create oracle
3. Create categorical event
4. Buy all outcomes

In the buy all outcomes will invoke 3 transactions (deposit, approve, buyAllTokens)

The "buyAllTokens" will fail if you don't increase the gasLimit a little.

Let me know if you need any further information!

Best regards,
Anxo

**Comment by cag** [comment]
Anxo,

Thank you for reporting this! I have reproduced the error, and it is as you have guessed: the gas limit estimation from the core contract layer does not match what is expected at least by MetaMask. With respect to your demo app, unsetting the gas limit default in the console via:

```js
App.gnosis.contracts.CategoricalEvent.defaults({gas:undefined})
```

delegates the gas estimation to MetaMask, which produces a limit of `300937`.

After the `buyAllOutcomes` transaction goes through, Ganache says the following about it:

```
  Transaction: 0x0a7be339a0708f06dbc7b36ed4be1573d66f0ccc68d7ba7eba118d15dbdc3028
  Gas usage: 200625
  Block Number: 49
  Block Time: Mon Feb 26 2018 14:12:52 GMT-0600 (CST)
```

Both of these are more than the built-in default of `136569`, which is calculated as 1.5 times the maximum gas cost as reported by the core contracts gas measurements layer found in `App.gnosis.contracts.CategoricalEvent.gasStats`.

**Comment by cag** [comment]
Okay, this doesn't explain everything, but fixing this would somewhat alleviate this issue (except for the highest value of ~300k):

https://github.com/gnosis/gnosis-contracts/issues/90

**Comment by cag** [comment]
Feel free to reopen this if the issue was not solved by #100 and has nothing to do with #101
